### PR TITLE
Saved Sites: prune objects on app launch / fire button

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/1201493110486074/1204879487866164/f
Task/Issue URL: https://app.asana.com/0/1201493110486074/1204874751768176/f
### Steps to test this PR

_Sync Disabled - App open clearing_
- [ ] Open app and add some bookmarks
- [ ] Delete bookmark
- [ ] Bookmark should still be in the DB with deleted flag enabled, and title / url removed (use AppInspector)
- [ ] Force close the app, and open it again
- [ ] Bookmark should be done from the DB 

_Sync Disabled - Fire button_
- [ ] Open app and add some bookmarks
- [ ] Ensure Fire Button config is set to clear all data
- [ ] Delete bookmark
- [ ] Bookmark should still be in the DB with deleted flag enabled, and title / url removed (use AppInspector)
- [ ] Tap on the fire button
- [ ] Bookmark should be done from the DB

### CODE REVIEW

**Comment by malmstein** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3276** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3276" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3276?utm_source=stack-comment).

File: saved-sites/saved-sites-store/src/main/java/com/duckduckgo/savedsites/store/SavedSitesEntitiesDao.kt [hunk 1]
```diff
@@ -66,13 +66,15 @@ interface SavedSitesEntitiesDao {
...
 
-    @Query("update entities set deleted = 1, lastModified = :lastModified where entityId = :id")
+    @Query("update entities set title = '', url = '', deleted = 1, lastModified = :lastModified where entityId = :id")
     fun delete(
         id: String,
         lastModified: String = DatabaseDateFormatter.iso8601(),
     )
 
-    @Query("update entities set deleted = 1, lastModified = :lastModified where entityId != :bookmarksRoot AND entityId != :favoritesRoot")
+    @Query(
```
Comment (aitorvs) [inline saved-sites/saved-sites-store/src/main/java/com/duckduckgo/savedsites/store/SavedSitesEntitiesDao.kt:75]: Curious, why are not just deleting the entry in the db?

Comment (malmstein) [reply saved-sites/saved-sites-store/src/main/java/com/duckduckgo/savedsites/store/SavedSitesEntitiesDao.kt:75]: This is called from the DevSettings screen, so I don’t see why we wouldn’t. Thanks for spotting that @aitorvs .

**Review by cmonfortep — APPROVED** [review]
LGTM 👍 

tested and works as expected

**Comment by malmstein** [comment]
Thanks @cmonfortep

**Pull request merged**